internal/databases/mysql/stream_fetch_handler_test.go

```diff
@@ -0,0 +1,81 @@
+package mysql
+
+import (
+	"bytes"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/internal/ioextensions"
+	"github.com/wal-g/wal-g/internal/storages/memory"
+	"github.com/wal-g/wal-g/internal/storages/storage"
+	"github.com/wal-g/wal-g/testtools"
+	"github.com/wal-g/wal-g/utility"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestFetchBinlogs(t *testing.T) {
+	storage_, cutPoint := fillTestStorage()
+
+	folder := memory.NewFolder("", storage_)
+	objects, _, err := folder.ListFolder()
+
+	var startBinlog storage.Object
+	for _, object := range objects {
+		if strings.HasPrefix(object.GetName(), "mysql-bin-log.000018.lz4") {
+			startBinlog = object
+		}
+	}
+
+	assert.NotNil(t, startBinlog)
+	assert.NoError(t, err)
+	assert.Equal(t, len(objects), 4)
+
+	allowed := []string{"mysql-bin-log.000018", "mysql-bin-log.000019"}
+
+	for _, object := range objects {
+		binlogName := utility.TrimFileExtension(object.GetName())
+		data, exist := storage_.Load(object.GetName())
+
+		assert.True(t, exist)
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		timestamp, err := parseFirstTimestampFromHeader(getTestReadSeekCloserWithExpectedData(mockCtrl, data.Data))
+
+		assert.NoError(t, err)
+		if internal.LogFileShouldBeFetched(startBinlog.GetLastModified(), nil, object) && !binlogIsTooOld(time.Unix(int64(timestamp), 0), &cutPoint) {
```

Comment:
I guess it's possible to use internal.FetchLogs here. Because it's a dangerous way to duplicate soft logic in the test.